### PR TITLE
Add Path.Join string overloads

### LINKED ISSUE #30049: Add string-based alternatives for Path.Join
In #25536 we've added 

```csharp
namespace System.IO
{
    public static class Path
    {
        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2);
        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, ReadOnlySpan<char> path3);
    }
}
```

The intent was that people will be able to call these APIs when they have a span or a string (because `string` has an implicit conversion to `ReadOnlySpan<char>`). However, some languages, such as F#, do not support implicit conversions. Also, it forces developers to know that this conversion exists. Normally, this isn't a big deal as developers learn common conversion between primitives but we don't expect the vast majority of developers having to use spans much -- it's meant as a low-level building block. Thus, they will largely remain unaware of the existence of that conversion.

So, to call these APIs from F# developers have write this:

```ml
Path.Join(p.AsSpan(), "obj".AsSpan())
```

To address these issues, we should just add `string`-based overloads which simply call the the other APIs:

```csharp
namespace System.IO
{
    public static class Path
    {
        // Existing
        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2);
        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2, ReadOnlySpan<char> path3);

        // New
        public static string Join(string path1, string path2);
        public static string Join(string path1, string path2, string path3);
    }
}
```

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
closes #30049 

/cc @JeremyKuhne

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@danmosemsft i don't understand why doesn't compile on some legs.
Another question, my PR merged 14 Jun on coreclr, i followed @dotnet-maestro-bot on corefx and on 16 Jun has updated CoreClr used runtime(downloaded on sync.cmd), but now on corefx i don't see new overloads https://github.com/dotnet/corefx/blob/master/src/Common/src/CoreLib/System/IO/Path.cs#L388
My question is...mirror folder and CoreClr runtime update are "unrelated"?

**Comment by JeremyKuhne** [comment]
UWP still needs the updated runtime bits.  Packaging failed for the same reason.  I'll kick it again.

```
00:25:49 D:\j\workspace\windows-TGrou---c60886e1\Tools\ApiCompat.targets(58,5): error : MembersMustExist : Member 'System.IO.Path.Join(System.String, System.String)' does not exist in the implementation but it does exist in the contract. [D:\j\workspace\windows-TGrou---c60886e1\src\System.Runtime.Extensions\src\System.Runtime.Extensions.csproj]
00:25:49 D:\j\workspace\windows-TGrou---c60886e1\Tools\ApiCompat.targets(58,5): error : MembersMustExist : Member 'System.IO.Path.Join(System.String, System.String, System.String)' does not exist in the implementation but it does exist in the contract. [D:\j\workspace\windows-TGrou---c60886e1\src\System.Runtime.Extensions\src\System.Runtime.Extensions.csproj]
```

> My question is...mirror folder and CoreClr runtime update are "unrelated"?

The shared folder files get mirrored to CoreFX and CoreRT. CoreRT then gets mirrored into internal UWP (ProjectN). ProjectN is the UWP CLR. It can take a bit before the bits have flowed all the way through and CoreFX has picked up the updated binaries.

**Comment by MarcoRossignoli** [comment]
@JeremyKuhne thank's for explanation!

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs [hunk 1]
```diff
@@ -84,6 +85,7 @@ public void TryJoinTwoPaths(string path1, string path2, string expected)
         [Theory, MemberData(nameof(TestData_JoinThreePaths))]
         public void JoinThreePaths(string path1, string path2, string path3, string expected)
         {
+            Assert.Equal(expected, Path.Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan()));
```
Comment (stephentoub) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: We should also have tests that verify the behavior when nulls are passed in.

Comment (MarcoRossignoli) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: Ok i'll do It!

**Comment by MarcoRossignoli** [comment]
> You do not have to block on this. To make the CI green, you can add the APIs to the ApiCompatBaseline.uapaot.txt, like it is done here for example: https://github.com/dotnet/corefx/blob/master/src/System.Runtime/src/ApiCompatBaseline.uapaot.txt

Sometimes i get this also when debug some code with VS. This works also in local?

**Comment by MarcoRossignoli** [comment]
Thank's @ViktorHofer!

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: dont, let me know if are enough.

**Comment by MarcoRossignoli** [comment]
@dotnet-bot test Linux arm Release Build please
@dotnet-bot test Linux arm64 Release Build please
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test Tizen armel Debug Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Linux arm Release Build please
@dotnet-bot test Linux arm64 Release Build please
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test Tizen armel Debug Build please

File: src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests_Join.netcoreapp.cs]: Thanks. LGTM.

**Pull request merged**